src/lib/csv-export/index.ts

```diff
@@ -0,0 +1,73 @@
+import {ModelPluginInterface} from '../../interfaces';
+import {ModelParams} from '../../types';
+
+import * as path from 'path';
+import * as fs from 'fs/promises';
+
+import {ERRORS} from '../../util/errors';
+import {buildErrorMessage} from '../../util/helpers';
+const {InputValidationError} = ERRORS;
+
+export class CsvExportModel implements ModelPluginInterface {
+  params: object | undefined;
+  outputPath: string | undefined;
+
+  errorBuilder = buildErrorMessage(CsvExportModel);
+
+  headers: string[] = [];
+
+  async configure(params: object | undefined): Promise<ModelPluginInterface> {
+    if (params === undefined) {
+      throw new InputValidationError(
+        this.errorBuilder({message: 'Configuration data is missing'})
+      );
+    }
+
+    this.params = params;
+
+    if ('output-path' in params) {
+      this.outputPath = params['output-path'] as string;
+    } else {
+      throw new InputValidationError(
+        this.errorBuilder({message: 'output-path required'})
+      );
+    }
+
+    if ('headers' in params) {
+      this.headers = params['headers'] as string[];
+    }
+
+    return this;
+  }
+
+  async execute(inputs: ModelParams[]): Promise<ModelParams[]> {
+    if (this.outputPath) {
```

Comment:
I would prefer to be cautious on this one as if someone tries to misuse or misunderstand the api (not call configure first) it is nice and clear that they need to call configure first. I know IF framework defaults to do this, but if any bug got into IF framework this would give a nice clear message.